### PR TITLE
Backport 2.28: PSA storage format: exercise key

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Backport of https://github.com/Mbed-TLS/mbedtls/pull/5644. Rework of https://github.com/Mbed-TLS/mbedtls/pull/4444. Continuation of https://github.com/Mbed-TLS/mbedtls/pull/5635.

* Mostly, this is a straightforward rebase of #5644, except for updating the generated files at every commit.
* In "Don't try to perform operations when driver support is lacking", add support for MD2 and MD4, which don't exist in `development`.
* Additional commits to handle RC4: it was only working for 128-bit keys, only for one-shot operations. Fix it for multipart operations. Don't extend it to other key sizes: it would require significant work and there's no need since it's never worked and nobody ever asked for it.
* In the rework of `cipher_encrypt_alg_without_iv` → `cipher_alg_without_iv`, `development` had a bit more support for multipart than 2.28.

Status: needs CI to check some weird configs

### CODE REVIEW

File: tests/suites/test_suite_psa_crypto.data [hunk 1]
```diff
@@ -1541,23 +1541,27 @@ cipher_encrypt_fail:PSA_ALG_CBC_NO_PADDING:PSA_KEY_TYPE_AES:"2b7e151628aed2a6abf
...
 depends_on:PSA_WANT_ALG_ECB_NO_PADDING:PSA_WANT_KEY_TYPE_DES
-cipher_encrypt_alg_without_iv:PSA_ALG_ECB_NO_PADDING:PSA_KEY_TYPE_DES:"01020407080b0d0ec1c2c4c7c8cbcdce":"c78e2b38139610e3":"5d0652429c5b0ac7"
+cipher_alg_without_iv:PSA_ALG_ECB_NO_PADDING:PSA_KEY_TYPE_DES:"01020407080b0d0ec1c2c4c7c8cbcdce":"c78e2b38139610e3":"5d0652429c5b0ac7"
 
 PSA symmetric encrypt: 3-key 3DES-ECB, 8 bytes, good
 depends_on:PSA_WANT_ALG_ECB_NO_PADDING:PSA_WANT_KEY_TYPE_DES
-cipher_encrypt_alg_without_iv:PSA_ALG_ECB_NO_PADDING:PSA_KEY_TYPE_DES:"01020407080b0d0ec1c2c4c7c8cbcdce31323437383b3d3e":"c78e2b38139610e3":"817ca7d69b80d86a"
+cipher_alg_without_iv:PSA_ALG_ECB_NO_PADDING:PSA_KEY_TYPE_DES:"01020407080b0d0ec1c2c4c7c8cbcdce31323437383b3d3e":"c78e2b38139610e3":"817ca7d69b80d86a"
+
+PSA symmetric encrypt: 128-bit RC4, 16 bytes, good
```
Comment (gabor-mezei-arm) [inline tests/suites/test_suite_psa_crypto.data:1562]:
```suggestion
PSA symmetric encrypt: 128-bit ARC4, 16 bytes, good
```

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]
It looks good to me, except one typo.

**Review by gabor-mezei-arm — DISMISSED** [review]
LGTM

File: ChangeLog.d/psa-rc4.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix PSA cipher multipart operations using RC4. Previously, an IV was
```
Comment (mprse) [inline ChangeLog.d/psa-rc4.txt:2]: I saw the comment reading typo (`RC4` and `ARC4`) and that both are the same.
Just pointing out that in change log `RC4` is used and in implementation and comments `ARC4` name is used.

**Review by mprse — APPROVED** [review]
Left one comment, otherwise looks good.

File: ChangeLog.d/psa-rc4.txt (hunk 2 above)
Comment (gilles-peskine-arm) [reply ChangeLog.d/psa-rc4.txt]: I amended the last commit to also change that changelog entry.

I didn't bother renaming the file, the file name will disappear once we merge the changelog entries anyway.

**Review by mprse — APPROVED** [review]
LGTM

**Pull request merged**